File: backend/Spotly/Spotly/Models/Dokumentacija.cs
using System;
using System.Collections.Generic;

namespace Spotly.Models;

public partial class Dokumentacija
{
    public int Id { get; set; }

    public byte[]? Slika { get; set; }

    public int? KaznaId { get; set; }

    public int? ZahtjevId { get; set; }

    public virtual Kazna? Kazna { get; set; }

    public virtual Zahtjev? Zahtjev { get; set; }
}

File: frontend/app/src/main/java/org/foi/hr/air/spotly/data/Dokumentacija.kt
package org.foi.hr.air.spotly.data

import android.util.Base64
import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.descriptors.PrimitiveKind
import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
import kotlinx.serialization.descriptors.SerialDescriptor
import kotlinx.serialization.encoding.Decoder
import kotlinx.serialization.encoding.Encoder

object ByteArrayAsBase64Serializer : KSerializer<ByteArray> {
    override val descriptor: SerialDescriptor =
        PrimitiveSerialDescriptor("Base64ByteArray", PrimitiveKind.STRING)

    override fun serialize(encoder: Encoder, value: ByteArray) {
        encoder.encodeString(Base64.encodeToString(value, Base64.NO_WRAP))
    }

    override fun deserialize(decoder: Decoder): ByteArray {
        return Base64.decode(decoder.decodeString(), Base64.DEFAULT)
    }
}

@Serializable
data class Dokumentacija(
    val id: Int,
    @Serializable(with = ByteArrayAsBase64Serializer::class) val slika: ByteArray?,
    val kaznaId: Int?,
    val zahtjevId: Int?
)